#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_gcc4.3_support.dpatch by Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for GCC 4.3, see bug #417491

@DPATCH@

--- a/intern/elbeem/intern/simulation_object.cpp
+++ b/intern/elbeem/intern/simulation_object.cpp
@@ -15,6 +15,7 @@
 #include "solver_interface.h"
 #include "particletracer.h"
 #include "elbeem.h"
+#include <cstdlib>
 
 #ifdef _WIN32
 #else
--- a/intern/elbeem/intern/solver_interface.cpp
+++ b/intern/elbeem/intern/solver_interface.cpp
@@ -16,6 +16,7 @@
 #include "ntl_ray.h"
 #include "ntl_world.h"
 #include "elbeem.h"
+#include <cstdlib>
 
 
 
--- a/intern/elbeem/intern/solver_main.cpp
+++ b/intern/elbeem/intern/solver_main.cpp
@@ -11,6 +11,7 @@
 #include "solver_relax.h"
 #include "particletracer.h"
 #include "loop_tools.h"
+#include <cstdlib>
 
 /*****************************************************************************/
 /*! perform a single LBM step */
--- a/intern/elbeem/intern/solver_util.cpp
+++ b/intern/elbeem/intern/solver_util.cpp
@@ -10,6 +10,7 @@
 #include "solver_class.h"
 #include "solver_relax.h"
 #include "particletracer.h"
+#include <cstdlib>
 
 // MPT
 #include "ntl_world.h"
--- a/intern/elbeem/intern/utilities.cpp
+++ b/intern/elbeem/intern/utilities.cpp
@@ -8,6 +8,7 @@
  *****************************************************************************/
 
 
+#include <cstdlib>
 #include <iostream>
 #include <sstream>
 #ifdef WIN32
--- a/intern/ghost/intern/GHOST_WindowX11.cpp
+++ b/intern/ghost/intern/GHOST_WindowX11.cpp
@@ -37,6 +37,7 @@
 // For standard X11 cursors
 #include <X11/cursorfont.h>
 #include <X11/Xatom.h>
+#include <cstdio>
 
 // For obscure full screen mode stuuf
 // lifted verbatim from blut.
--- a/source/blender/yafray/intern/export_File.cpp
+++ b/source/blender/yafray/intern/export_File.cpp
@@ -1,6 +1,7 @@
 #include "export_File.h"
 
 #include <math.h>
+#include <cstring>
 
 using namespace std;
 
--- a/source/blender/yafray/intern/export_Plugin.cpp
+++ b/source/blender/yafray/intern/export_Plugin.cpp
@@ -1,6 +1,7 @@
 #include "export_Plugin.h"
 
 #include <math.h>
+#include <cstring>
 using namespace std;
 
 
--- a/source/blender/yafray/intern/yafray_Render.cpp
+++ b/source/blender/yafray/intern/yafray_Render.cpp
@@ -7,6 +7,7 @@
 #include "yafray_Render.h"
 
 #include <math.h>
+#include <cstring>
 
 using namespace std;
 
--- a/source/gameengine/BlenderRoutines/BL_KetsjiEmbedStart.cpp
+++ b/source/gameengine/BlenderRoutines/BL_KetsjiEmbedStart.cpp
@@ -663,4 +663,5 @@ extern "C" void StartKetsjiShellSimulation(struct ScrArea *area,
 
 	} while (exitrequested == KX_EXIT_REQUEST_RESTART_GAME || exitrequested == KX_EXIT_REQUEST_START_OTHER_GAME);
 	if (bfd) BLO_blendfiledata_free(bfd);
-}
\ No newline at end of file
+}
+
--- a/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/glext.h
+++ b/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/glext.h
@@ -54,7 +54,9 @@ extern "C" {
 /* Header file version number, required by OpenGL ABI for Linux */
 /* glext.h last updated 2005/06/20 */
 /* Current version at http://oss.sgi.com/projects/ogl-sample/registry/ */
+#ifndef GL_GLEXT_VERSION
 #define GL_GLEXT_VERSION 29
+#endif
 
 #ifndef GL_VERSION_1_2
 #define GL_UNSIGNED_BYTE_3_3_2            0x8032
